cla_backend/libs/eligibility_calculator/calculator.py

```diff
@@ -330,25 +348,309 @@ def is_disposable_capital_eligible(self):
+            if filter_to_keys(cfe_calcs, self.calcs) != self.calcs:
+                logger.error("CFE and legacy_check() calcs disagree!\nCFE:    %s\nLegacy: %s" % (filter_out_zeros(cfe_calcs), filter_out_zeros(self.calcs)))
+
+        # Gradual cut-over from using legacy_result to cfe_result
+        if self._is_non_means_tested(self.case_data) or self._without_partner(self.case_data):
+
+            # Calcs updated from CFE's result
+            self.calcs = cfe_calcs
+
+            logger.info("Eligibility result (using CFE): %s", cfe_result)
+            return cfe_result
+        else:
+            logger.info("Eligibility result (using legacy): %s", legacy_result)
+            return legacy_result
+
+    @staticmethod
+    def _pounds_to_pence(value):
+        # deal with rounding error by adding a small amount before truncating
+        return int(value * 100 + 0.1)
+
+    @staticmethod
+    def _without_partner(case_data):
+        return hasattr(case_data.facts, "has_partner") and not case_data.facts.has_partner
+
+    @staticmethod
+    def _is_non_means_tested(case_data):
+        return case_data.facts.on_nass_benefits and case_data.category == "immigration"
+
+    def _do_cfe_civil_check(self):
+        if not EligibilityChecker._is_applicant_detail_section_complete(self.case_data):
+            # data is so incomplete that we can't even call CFE sensibly
+            result = ELIGIBILITY_STATES.UNKNOWN
+            logger.info("Eligibility result (CFE): %s %s" % (result, "couldn't call CFE"))
+            return result, None, None
+
+        cfe_request_dict = self._translate_case(self.case_data)
+
+        cfe_raw_response = requests.post(settings.CFE_URL, json=cfe_request_dict)
+        logger.debug("Eligibility request (CFE): %s" % json.dumps(cfe_request_dict, indent=4, sort_keys=True))
+
+        cfe_response = CfeResponse(cfe_raw_response.json())
+        result, calcs = self._translate_response(cfe_response)
+        logger.debug(
+            "Eligibility result (CFE): %s" % (json.dumps(cfe_response._cfe_data, indent=4, sort_keys=True)))
+
+        return result, calcs, cfe_response
+
+    @staticmethod
+    def _is_applicant_detail_section_complete(case_data):
+        return hasattr(case_data.facts, "dependants_young") and hasattr(case_data.facts, "has_partner")
+
+    @staticmethod
+    def _translate_case(case_data, submission_date=None):
+        '''Translates CLA's CaseData to CFE-Civil request JSON'''
+        if not submission_date:
+            submission_date = datetime.date.today()
+        # produce the simplest possible plain request to CFE to prove the route
+        request_data = {
+            "assessment": {
+                "submission_date": str(submission_date),
+                "level_of_help": "controlled"  # CLA is for 'advice' only, so always controlled
+            },
+            "applicant": {
+                "date_of_birth": "1992-07-25",
```

Comment:
Are these hard-coded here because they are the minimum amount of data CFE requires and CLA does not capture them?

Reply:
These are default values, and get overwritten by the translators (`translate_applicant_data` and `translate_proceeding_types`), which are called from later in this method.

There's an argument we don't need to have these defaults. They would only be used if the CaseData was missing the "facts" and "category" keys. There is a check elsewhere for facts.has_partner, and I'm confident the "category" is always going to be there too, but it is reassuring to have a working CFE request to start with.

Reply:
Ah I see @starswan has removed applicant section in [12f3b3a](https://github.com/ministryofjustice/cla_backend/pull/1082/commits/12f3b3a4adce3a039281a1a4c9e79976437cc69a) which is also fine